lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -1790,17 +1793,123 @@ class _DomTouchListWrapper<T> extends Iterable<T> {
 Iterable<T> createDomTouchListWrapper<T>(_DomTouchList list) =>
     _DomTouchListWrapper<T>._(list).cast<T>();
 
+@JS()
+@staticInterop
+class DomSymbol {}
+
+extension DomSymbolExtension on DomSymbol {
+  external Object get iterator;
+}
+
 @JS()
 @staticInterop
 class DomIntl {}
 
 extension DomIntlExtension on DomIntl {
+  // ignore: non_constant_identifier_names
+  external Object? get Segmenter;
+
   /// This is a V8-only API for segmenting text.
   ///
   /// See: https://code.google.com/archive/p/v8-i18n/wikis/BreakIterator.wiki
   external Object? get v8BreakIterator;
 }
 
+/// Similar to [js_util.callMethod] but allows for providing a non-string method
+/// name.
+T jsCallMethod<T>(Object o, Object method, [List<Object?> args = const <Object?>[]]) {
+  final Object actualMethod = js_util.getProperty(o, method);
+  return js_util.callMethod<T>(actualMethod, 'apply', <Object?>[o, args]);
```

Comment:
Let's check with @joshualitt if this is future-proof.

Reply:
I'm committed to migrating the engine shim to whatever future interop looks like. That said, I'm hoping to have a replacement for js_util based on JS types, and when I do I can migrate this use case.